### PR TITLE
Add learnset targeting to /movesearch

### PR DESCRIPTION
**Pull request by SolarisFox**
Added a parameter to /movesearch that only searches moves from a target pokemon's learnset.

### CODE REVIEW

**Comment by kotarou3** [comment]
Run `npm test` _before_ submitting the pull...

**Comment by Slayer95** [comment]
Why not have this as part of /movesearch functionality?
I don't think its heuristics would get along well with the original exact /learn command.

File: config/commands.js [hunk 1]
```diff
@@ -1100,24 +1100,67 @@ var commands = exports.commands = {
...
-				return this.sendReply("Move '" + move.id + "' not found.");
+				var matchedMoves = [];
+				var targetType = '';
+				var lset = template.learnset;
+				target = targets[i].trim().toLowerCase();
+
+				var typeIndex = target.indexOf(" type");
+				if (typeIndex > -1) {
+					targetType = target.substr(0, typeIndex).capitalize();
+					if (!(targetType in Tools.data.TypeChart)) return this.sendReply("Type '" + Tools.escapeHTML(targetType) + "' not found.");
```
Comment (Slayer95) [inline config/commands.js]: Use <code>Tool.getType</code> and check <code>.exists</code> rather than sugarjs' <code>capitalize</code> and <code>in</code>.

**Comment by SolarisFox** [comment]
Hmm, yeah that probably makes more sense. @QuiteQuiet recommended this be added to /learn, which made sense to me at the time but I'll change it over to be part of /movesearch.

**Comment by SolarisFox** [comment]
I'm not really sure how to make the output most concise. I liked the way the indentation looked when using  ```<ul class="message-learn-list">``` however I'm not sure if that would make as much sense here as it would be inconsistent with the output from searching from the entire movedex. To be completely consistent I would not return ```"Matching moves found in learnset for " + targetMon + ":``` at all, but if multiple searches were run in a row trying to find a Pokemon that has what they need, it would help to read back through the searches.

**Comment by SolarisFox** [comment]
Fixed conflict created with #1742 

Also added in two extra whitespace lines to match /dexsearch

**Pull request merged**